### ISSUE TITLE
Unexpected start times reported for skipped steps

### ISSUE THREAD

**Issue by AlanGreene** [tektoncd/pipeline#2416]
# Expected Behavior
Step start times should be sequential regardless of whether they actually ran or not.

# Actual Behavior
If a TaskRun fails, any remaining skipped steps retain their original container start time instead of being updated by the entrypoint binary.

# Steps to Reproduce the Problem

```
kubectl create -f - << EOF
---
apiVersion: tekton.dev/v1beta1
kind: TaskRun
metadata:
  generateName: test-start-time-correct-
spec:
  taskSpec:
    steps:
    - image: ubuntu
      script: |
        #!/usr/bin/env bash
        sleep 3 && echo step 1
    - image: ubuntu
      script: |
        #!/usr/bin/env bash
        sleep 3 && echo step 2
    - image: ubuntu
      script: |
        #!/usr/bin/env bash
        sleep 3 && echo step 3
    - image: ubuntu
      script: |
        #!/usr/bin/env bash
        sleep 3 && echo step 4
---
apiVersion: tekton.dev/v1beta1
kind: TaskRun
metadata:
  generateName: test-start-time-wrong-
spec:
  taskSpec:
    steps:
    - image: ubuntu
      script: |
        #!/usr/bin/env bash
        sleep 3 && echo step 1
    - image: ubuntu
      script: |
        #!/usr/bin/env bash
        sleep 3 && echo step 2
        false
    - image: ubuntu
      script: |
        #!/usr/bin/env bash
        sleep 3 && echo step 3
    - image: ubuntu
      script: |
        #!/usr/bin/env bash
        sleep 3 && echo step 4
EOF
```

The step start times for `test-start-time-correct-` are recorded and reported correctly, reflecting the time the step actually executed.

The step start times for the first 2 steps in `test-start-time-wrong-` are correct, but the last 2 skipped steps still report their original container start times. This means that if I order the step status by start time they appear before the steps that actually ran.

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

  ```
  Client Version: v1.16.3
  Server Version: v1.15.5
  ```

- Tekton Pipeline version:

  **Output of `tkn version` or `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`**

  v0.11.1

/kind bug

Related issue: https://github.com/tektoncd/pipeline/issues/1512
This fixed the start times for steps that are actually executed.

**Comment by vincent-pli** [comment]
Fixed by #2429

**Comment by AlanGreene** [comment]
Thanks @vincent-pli, verified on the latest nightly build.
